components/feature/session/src/main/java/mozilla/components/feature/session/PictureInPictureFeature.kt

```diff
@@ -39,7 +42,15 @@ class PictureInPictureFeature(
             sessionManager.runWithSessionIdOrSelected(customTabSessionId) { session ->
                 session.fullScreenMode
             }
-        return fullScreenMode && enterPipModeCompat()
+        return fullScreenMode && try {
+            enterPipModeCompat()
+        } catch (e: IllegalStateException) {
+            // On certain Samsung devices, if accessibility mode is enabled, this will throw an
+            // IllegalStateException even if we check for the system feature beforehand. So let's
+            // catch it, log it, and not enter PiP. See https://stackoverflow.com/q/55288858
+            logger.error("Entering PipMode failed", e)
```

Comment:
The logger would add the error to the on-device logs only. We should pass in `CrashReporting` instance here so that we can log it on one of our crash reporters. Here is an [example usage](https://searchfox.org/mozilla-mobile/source/android-components/components/support/migration/src/main/java/mozilla/components/support/migration/FennecMigrator.kt#653).